spacy/util.py

```diff
@@ -1032,9 +1033,18 @@ def make_tempdir() -> Generator[Path, None, None]:
     YIELDS (Path): The path of the temp directory.
     """
     d = Path(tempfile.mkdtemp())
```

Comment:
Maybe this is a carry-over from python 2 and we should switch to `tempfile.TemporaryDirectory` or similar?

Reply:
But then we'd always leave junk around on Windows since `tempfile.TemporaryDirectory` only seems to allow ignoring errors, not providing a function to override the behavior? Or maybe I missed it.

Reply:
Hmm, I had some hope that it might actually work correctly?

Reply:
Cool, it seems to do so:

https://github.com/python/cpython/blob/498598e8c2d64232d26c075de87c513415176bbf/Lib/tempfile.py#L867

I'll try it in the other PR.

Reply:
If it works in the other PR, let's move those changes to a separate PR?

Reply:
It looks like it works in Python 3.8 and higher, and <=3.7 will need a workaround:

https://github.com/python/cpython/issues/70847

Reply:
ugh

Reply:
The change is small, so I've modified it to use TemporaryDirectory so we can check it works as documented here as well. If we do it in another PR we can just merge it in here.

Reply:
Didn't work in the other PR due to the issue that you pointed out above (<=3.7). So, do we want the uglier version or maybe target spaCy v4, since we are probably going to drop 3.7?

Reply:
This is already targeting v4, actually, so I suggest we just leave it as it is now. (The tests also confirmed it works except for 3.7 Windows.)